src/components/button/button-types.style.js

```diff
@@ -9,92 +9,93 @@ export function makeColors(color) {
   `;
 }
 
-export default ({ colors, disabled }, isDisabled, destructive) => ({
-  primary: `
-    background: ${colors.primary};
-    border-color: transparent;
-    color: ${colors.white};
-    &:hover {
-      background: ${colors.secondary};
-    }
+const IF = (bool, css) => (bool ? css : '');
 
-    ${isDisabled ? `
-    background: ${disabled.button};
-    color: ${disabled.text};
-    &:hover {
-      background: ${disabled.button};
-      border-color: ${disabled.button};
-      color: ${disabled.text};
-    }
-  ` : ''}
-
-    ${destructive ? `background: ${colors.error};
-    border-color: transparent;
-    color: ${colors.white};
-    &:hover {
-      background: ${colors.destructive.hover};
-    }
+export default ({
+  theme: { colors, disabled }, disabled: isDisabled, destructive, buttonType
+}) => {
+  const styles = {
+    primary: `
+      background: ${colors.primary};
+      border-color: transparent;
+      color: ${colors.white};
+      &:hover {
+        background: ${colors.secondary};
+      }
+  
+      ${IF(destructive, `
+      background: ${colors.error};
+      border-color: transparent;
+      color: ${colors.white};
```

Comment:
that does nothing, as the border color is already transparent and text is already white